### PR TITLE
Feat/alert dialog wrong blockchain

### LINKED ISSUE #23: Add Alert Dialog to make sure User is connected to Fuji

### PR DESCRIPTION
**Pull request by yvesbou**
- [x] check on visit if correct blockchain (fuji)
- [x] check on chain changes if correct blockchain (fuji)
- [x] if connected to wrong blockchain, block application by showing alert dialog 
- [x] prompt the user to change to fuji via option on dialog
- [x] if the user doesn't have fuji added to metamask, use RPC API to add it

@leovct it is a branch off from snackbar, should we merge into snackbar or main?

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -55,6 +57,92 @@ export default function App() {
...
+
+      if (!ethereum) {
+        alert("Get Metamask!");
+        return;
+      }
+
+      ethereum.on('chainChanged', (chainId) => {
+        console.log(chainId);
+        // Handle the new chain.
+        if (!(chainId === 43113 || chainId === 43114)) {
```
Comment (leovct) [inline src/App.js]: We shouldn't hardcode any data (here the id of the fuji and avalanche chains).
Instead, we should create a `constants` directory under `src` and a file `address.js` where we store these values :
```js
export const FUJI_TESTNET_ID = 43113;
export const AVALANCHE_MAINNET_ID = 43114;
```

Comment (leovct) [inline src/App.js]: We shouldn't hardcode any data (here the id of the fuji and avalanche chains).
Instead, we should create a `constants` directory under `src` and a file `address.js` where we store these values :
```js
export const FUJI_TESTNET_ID = 43113;
export const AVALANCHE_MAINNET_ID = 43114;
```

File: src/App.js [hunk 2]
```diff
@@ -55,6 +57,92 @@ export default function App() {
...
+    try {
+      const { ethereum } = window;
+      if (!ethereum) {
+        alert("Get Metamask!");
+        return;
+      }
+      // 0xA869 == 43113 (Fuji)
+      await window.ethereum.request({
+        method: 'wallet_switchEthereumChain',
+        params: [{ chainId: '0xA869' }],}); // chainId must be in hexadecimal numbers
```
Comment (leovct) [inline src/App.js]: Same thing here :)

Comment (leovct) [inline src/App.js]: Same thing here :)

File: src/App.js [hunk 3]
```diff
@@ -55,6 +57,92 @@ export default function App() {
...
+
+    } catch (switchError) {
+      //error code (error.code) is 4902, then the requested chain has not been added by MetaMask
+      if (switchError.code === 4902) {
+        try {
+          await window.ethereum.request({
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainId: '0xA869',
```
Comment (leovct) [inline src/App.js]: Same thing here (chainId, rpcUrl, blockExplorerUrl, etc.)

Comment (leovct) [inline src/App.js]: Same thing here (chainId, rpcUrl, blockExplorerUrl, etc.)

**Comment by leovct** [comment]
> * [x]  check on visit if correct blockchain (fuji)
> * [x]  check on chain changes if correct blockchain (fuji)
> * [x]  if connected to wrong blockchain, block application by showing alert dialog
> * [x]  prompt the user to change to fuji via option on dialog
> * [x]  if the user doesn't have fuji added to metamask, use RPC API to add it
> 
> @leovct it is a branch off from snackbar, should we merge into snackbar or main?

We'll merge the snackbar PR first and then merge this PR :)

**Comment by yvesbou** [comment]
to-do before merging:
- [x] constants directory 
- [x] a file called address.js with all the information about fuji/mainnet
- [x] strings with variables from constants directory

**Review by leovct — APPROVED** [review]

**Pull request merged**